### DISCUSSION TITLE
pkg_resources deprecation timeline

### DISCUSSION THREAD

**Discussion by feanil** [pypa/setuptools#3934] [Q&A] [answered]
I know that pkg_resources is [deprecated](https://setuptools.pypa.io/en/latest/pkg_resources.html) and will be removed in some future version.  Is there a sense of what the timeline for removal looks like?

**Comment by abravalheri** [comment, marked as answer]
Hi @feanil, right now, we are not yet in a position to estimate a timeline for the removal of `pkg_resources`. It is a complicated piece of software and it will take time to disentangle the codebase.

Our intent in declaring the deprecation was to communicate with the users know as soon as possible about the future change, so they can start making the necessary changes as soon as they can.

**Reply by feanil** [reply]
Makes sense, I assume as you get closer there will be more announcements but I just wanted to make sure I didn't miss this info if it was known.

**Reply by stevepiercy** [reply]
@abravalheri is there a status or timeline update available? If we know, we can better plan our releases. Ref: https://github.com/plone/Products.CMFPlone/issues/3928#issuecomment-2646629353. Thank you!

**Reply by abravalheri** [reply]
Hi @stevepiercy, there is not.
